pkg/node/node.go

```diff
@@ -46,280 +43,91 @@ func logErr(e error) error {
-			}
-			return name, namespace, nil
-		}
-	}
-
-	return "", "", nil
-}
-
-func (n NodeServer) NodeUnstageVolume(ctx context.Context, request *csi.NodeUnstageVolumeRequest) (*csi.NodeUnstageVolumeResponse, error) {
-	panic("implement me")
-}
-
-func (n NodeServer) NodePublishVolume(ctx context.Context, request *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
-	vID := request.GetVolumeId()
-	stagingTargetPath := request.GetStagingTargetPath()
-	targetPath := request.GetTargetPath()
-
-	if vID == "" {
-		return nil, status.Error(codes.InvalidArgument, "volume ID missing in request")
-	}
-
-	if err := os.MkdirAll(targetPath, 0755); err != nil {
-		return nil, status.Errorf(codes.Internal, "Stage Volume Failed: %v", err)
-	}
-
-	if err := mount.New("").Mount(stagingTargetPath, targetPath, "", []string{"bind"}); err != nil {
-		return nil, status.Errorf(codes.Internal, "Publish Volume Mount Failed: %v", err)
+		return nil, err
 	}
 
-	return &csi.NodePublishVolumeResponse{}, nil
-}
-
-func (n NodeServer) NodeUnpublishVolume(ctx context.Context, request *csi.NodeUnpublishVolumeRequest) (*csi.NodeUnpublishVolumeResponse, error) {
-	klog.Infof("NodeUnpublishVolume: volId: %v, targetPath: %v\n", request.GetVolumeId(), request.GetTargetPath())
-	target := filepath.Join(request.TargetPath, protocolFileName)
-	err := os.RemoveAll(target)
+	err = n.provisioner.removeMount(request.GetTargetPath())
 	if err != nil {
-		if os.IsNotExist(err) {
-			return nil, nil
-		}
-		return nil, logErr(fmt.Errorf("unable to remove file %s: %v", target, err))
+		return nil, err
 	}
 	return &csi.NodeUnpublishVolumeResponse{}, nil
 }
 
-func (n NodeServer) NodeGetVolumeStats(ctx context.Context, request *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	panic("implement me")
-}
-
-func (n NodeServer) NodeExpandVolume(ctx context.Context, request *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
-	panic("implement me")
+func (n *NodeServer) NodeGetInfo(ctx context.Context, request *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
+	resp := &csi.NodeGetInfoResponse{
+		NodeId:            n.nodeID,
+		MaxVolumesPerNode: n.volumeLimit,
+	}
+	return resp, nil
 }
 
-func (n NodeServer) NodeGetCapabilities(ctx context.Context, request *csi.NodeGetCapabilitiesRequest) (*csi.NodeGetCapabilitiesResponse, error) {
-	panic("implement me")
+func (n *NodeServer) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetCapabilitiesRequest) (*csi.NodeGetCapabilitiesResponse, error) {
+	return &csi.NodeGetCapabilitiesResponse{}, nil
 }
 
-func (n NodeServer) NodeGetInfo(ctx context.Context, request *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
-	klog.Infof("NodeGetInfo()")
-	resp := &csi.NodeGetInfoResponse{
-		NodeId: n.nodeID,
+func parseData(s *v1.Secret) (map[string]string, error) {
+	output := make(map[string]string)
+	for key, value := range s.Data {
```

Comment:
why can't you simply return `s.Data`?

Reply:
When I tried to return s.Data before, the value of the map (which is a []byte) did not marshal correctly and it was not valid JSON. I can try this again to validate the behaviour.

Reply:
Update, just tried:
s.Data -> {"key":"eyJzZWNyZXQiOiJwYXNzd29yZCJ9Cg=="}
Convert to map[string]string -> {"key":"{\"secret\":\"password\"}\n"}